### PR TITLE
feat(resource): deprecate `auto_restart_interval` in favor of `health_check_interval`

### PR DESCRIPTION
**Pull request by thalesmg**
See:
https://emqx.atlassian.net/wiki/spaces/P/pages/612368639/open+e5.1+remove+auto+restart+interval+from+buffer+worker+resource+options

Current problem:

In 5.0.x, we have two timer options that control the state changing of buffer worker
resources: auto_restart_interval and health_check_interval.

- auto_restart_interval controls how often the resource attempts to transition from
disconnected to connected.

- health_check_interval controls how often the resource is checked and potentially moved
from connected to disconnected or connecting.

The existence of two independent timers for very similar purposes is confusing to users,
QA and even developers.  Also, an intimately related configuration is request_timeout,
which can interact badly with auto_restart_interval if the latter is poorly configured:
requests may always expire if request_timeout < auto_restart_interval and if the resource
enters the disconnected state.  For health_check_interval, we attempt to derive a sane
default that gives requests a chance to retry (if request timeout is finite, then the
resource retries requests with a period of min(health_check_interval, request_timeout /
3).

Another problem with the separate auto_restart_interval is that its default value (60 s)
is too high when compared to the default request timeout and health check, leading to the
problems described above if not tuned.

Proposed solution:

We propose to drop auto_restart_interval in favor of health_check_interval, which will be
used for both disconnected -> connected and connected -> {disconnected, connecting}
transition checks.  With that, the resource will attempt to reconnect at the same interval
as the health check, which currently is 15 s.

Also, as two smaller changes to accompany this one:

- Increase the default request_timeout from 15 s to 45 s.
- Rename request_timeout to request_ttl.

### <samp>🤖 Generated by Copilot at 34d1c9e</samp>

Refactor the bridge resource management and configuration options to improve consistency, clarity, and simplicity. Remove the `auto_restart_interval` option and feature from all bridge types, as it is redundant and problematic. Rename the `request_timeout` option to `request_ttl` in all bridge modules and test suites, to avoid naming conflicts and unify the meaning of the request time-to-live option. Bump the version numbers of the affected bridge applications to reflect the changes.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_resource/src/schema/emqx_resource_schema.erl [hunk 1]
```diff
@@ -124,39 +122,22 @@ start_timeout(required) -> false;
...
 query_mode(desc) -> ?DESC("query_mode");
 query_mode(default) -> async;
 query_mode(required) -> false;
 query_mode(_) -> undefined.
 
-request_timeout(type) -> hoconsc:union([infinity, emqx_schema:duration_ms()]);
-request_timeout(desc) -> ?DESC("request_timeout");
-request_timeout(default) -> <<"15s">>;
-request_timeout(_) -> undefined.
+request_ttl(type) -> hoconsc:union([infinity, emqx_schema:duration_ms()]);
```
Comment (zmstone) [inline apps/emqx_resource/src/schema/emqx_resource_schema.erl:136]: keep the default value's type as the first union member.
```suggestion
request_ttl(type) -> hoconsc:union([emqx_schema:duration_ms(), infinity]);
```

**Review by zmstone — COMMENTED** [review]
please add a changelog to describe in detail the changed fields and default values.

**Comment by thalesmg** [comment]
> please add a changelog to describe in detail the changed fields and default values.

:heavy_check_mark:

**Review by savonarola — APPROVED** [review]

**Comment by yanzhiemq** [comment]
_

**Comment by yanzhiemq** [comment]
_

**Pull request merged**